### PR TITLE
build: Adds a envoy_select_signal_trace Bazel macro

### PR DESCRIPTION
**Pull request by abeyad**

### CODE REVIEW

File: bazel/envoy_select.bzl [hunk 1]
```diff
@@ -108,6 +108,13 @@ def envoy_select_hot_restart(xs, repository = ""):
         "//conditions:default": xs,
     })
 
+# Selects the given values if signal trace is enabled in the current build.
+def envoy_select_signal_trace(xs, repository = ""):
+    return select({
+        repository + "//bazel:disable_signal_trace": [],
```
Comment (RyanTheOptimist) [inline bazel/envoy_select.bzl:114]: Shall we add the mobile-specific logic here too, or do you want to do so in a followup?

File: mobile/library/common/BUILD [hunk 2]
```diff
@@ -53,12 +53,10 @@ envoy_cc_library(
...
         "@envoy//source/exe:stripped_main_base_lib",
-    ] + select({
-        "@envoy//bazel:disable_signal_trace": [],
-        "//conditions:default": [
-            "@envoy//source/common/signal:sigaction_lib",
-        ],
-    }),
+    ] + envoy_select_signal_trace(
+        ["@envoy//source/common/signal:sigaction_lib"],
+        "@envoy",
```
Comment (RyanTheOptimist) [inline mobile/library/common/BUILD:58]: Out of curiosity, is @envoy needed here?

File: mobile/test/common/integration/BUILD [hunk 3]
```diff
@@ -159,12 +160,10 @@ envoy_cc_test_library(
...
         "@envoy_build_config//:extension_registry",
-    ] + select({
-        "@envoy//bazel:disable_signal_trace": [],
-        "//conditions:default": [
-            "@envoy//source/common/signal:sigaction_lib",
-        ],
-    }),
+    ] + envoy_select_signal_trace(
+        ["@envoy//source/common/signal:sigaction_lib"],
+        "@envoy",
```
Comment (RyanTheOptimist) [inline mobile/test/common/integration/BUILD:165]: ditto

**Review by RyanTheOptimist — COMMENTED** [review]
Looks great! Thanks for doing this. A couple of minor questions.

File: mobile/test/common/integration/BUILD (hunk 3 above)
Comment (abeyad) [reply mobile/test/common/integration/BUILD:165]: Indeed it is, for the reasons mentioned above

File: mobile/library/common/BUILD (hunk 2 above)
Comment (RyanTheOptimist) [reply mobile/library/common/BUILD:58]: OH! Because this target is in mobile/ which is not part of envoy! Ah! Thanks!

**Review by RyanTheOptimist — DISMISSED** [review]

**Review by RyanTheOptimist — APPROVED** [review]

**Comment by abeyad** [comment]
added @keith for cross-company review

File: .bazelrc [hunk 4]
```diff
@@ -262,7 +262,7 @@ build:remote --strategy=Genrule=remote,sandboxed,local
 build:remote --remote_timeout=7200
 build:remote --google_default_credentials=true
 build:remote --remote_download_toplevel
-build:remote --nobuild_runfile_links
+# build:remote --nobuild_runfile_links
```
Comment (keith) [inline .bazelrc:265]: unintentional check in?

**Review by keith — APPROVED** [review]
looks good!

**Review by keith — APPROVED** [review]

File: .bazelrc (hunk 4 above)
Comment (abeyad) [reply .bazelrc]: undid this just now, sorry for the confusion!  i was just trying out undoing a phlax change that was causing CI to fail.  phlax is going to send in a fix for it tomorrow.  thanks for reviewing!

**Comment by RyanTheOptimist** [comment]
Now that phlax's PR has landed, I think this just needs a main merge.

**Pull request merged**